lib/apollo-federation/entities_field.rb

```diff
@@ -49,6 +49,14 @@ def _entities(representations:)
         # TODO: What if the type is an interface?
         type_class = class_of_type(type)
 
+        if type_class.underscore_reference_keys
+          representations.each do |representation|
+            representation.transform_keys! do |key|
+              GraphQL::Schema::Member::BuildType.underscore(key.to_s).to_sym
+            end
+          end
```

Comment:
We might want to transform a copy instead of the original representations. Could cause weird issues down the road if we do this destructively.